zfsbootmenu/lib/zfsbootmenu-core.sh

```diff
@@ -2272,3 +2272,40 @@ is_zfs_filesystem() {
 
   return 1
 }
+
+# arg1: mount point
+# returns: 0 if everything was unmounted, 1 if not
+
+recursive_umount() {
+  local mounts mountpoint mp skip mp_re depth filesystem ret
+
+  mountpoint="${1}"
+  if [ -z "${mountpoint}" ]; then
+    zerror "mountpoint undefined"
+    return 1
+  fi
+
+  ret=0
+  if ! umount -R "${mountpoint}" >/dev/null 2>&1 ; then
```

Comment:
```suggestion
  umount -R "${mountpoint}" >/dev/null 2>&1 && return
  
  ret=0
```
Then you can drop the big `if` indent.